File: src/main/java/practice/level01/arrays/MinStepsInInfiniteGrid/Solution.cs
namespace practice.level01.MinStepsInInfiniteGrid
{
    using System;
    using System.Collections.Generic;
    using System.Linq;

    public class Solution
    {
        public int coverPoints(List<int> A, List<int> B)
        {
            var result = 0;
            if (A.Count() < 2 && B.Count() < 2)
                return result;
            result = Enumerable.Range(1, A.Count() - 1)
                .Select(index =>
                {
                    var diffInFirstArray = Math.Abs(A[index] - A[index - 1]);
                    var diffInSecondArray = Math.Abs(B[index] - B[index - 1]);
                    return diffInFirstArray > diffInSecondArray ? diffInFirstArray : diffInSecondArray;

                })
                .Sum();
            return result;
        }
    }
}

File: src/main/java/practice/level01/arrays/MinStepsInInfiniteGrid/solution.swift
import Foundation

class Solution {
    func coverPoints(_ A: inout [Int], _ B: inout [Int]) -> Int {
        var result = 0
        if (A.count < 2 && B.count < 2) {
            return result
        }
        for i in 1...A.count - 1 {
            var diffFirstArray = abs(A[i] - A[i - 1])
            var diffSecondArray = abs(B[i] - B[i])
            result += diffFirstArray > diffSecondArray ? diffFirstArray : diffSecondArray
        }
        return result
    }
}
